cdap-data-fabric/src/main/java/co/cask/cdap/data2/transaction/queue/hbase/HBaseConsumerStateStore.java

```diff
@@ -368,6 +371,15 @@ private void deleteRemovedGroups(Row row, Set<Long> existingGroups) {
     return scanBarriers(groupId, new AllCollector<QueueBarrier>()).finish(new ArrayList<QueueBarrier>());
   }
 
+  /**
+   * Gets all barrier information for all groups. The information are sorted in the order of
+   * the barrier changes.
+   */
+  LinkedHashMultimap<Long, QueueBarrier> getAllBarriers() {
```

Comment:
The return type can still be `Multimap`. Just the implementation is `LinkedHashMultimap` to guarantee the ordering (there is other implementation can do that, that's why the return type should be just Multimap).